handlers/infobip/infobip.go

```diff
@@ -1,3 +1,124 @@
 package infobip
 
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+)
+
 /* no logs */
+
+func init() {
+	courier.RegisterHandler(NewHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+// NewHandler returns a new Infobip handler
+func NewHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("IB"), "Infobip")}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	err := s.AddReceiveMsgRoute(h, "POST", "receive", h.ReceiveMessage)
+	return err
+}
+
+// ReceiveMessage is our HTTP handler function for incoming messages
+func (h *handler) ReceiveMessage(channel courier.Channel, w http.ResponseWriter, r *http.Request) ([]courier.ReceiveEvent, error) {
+	ie := &infobipEnvelope{}
+	err := handlers.DecodeAndValidateJSON(ie, r)
+	if err != nil {
+		return nil, courier.WriteError(w, r, err)
+	}
+
+	if ie.MessageCount == 0 {
+		return nil, courier.WriteIgnored(w, r, "ignoring request, no message")
+	}
+
+	msgs := []courier.Msg{}
+	for _, infobipMessage := range ie.Results {
+		messageID := infobipMessage.MessageID
+		text := infobipMessage.Text
+		dateString := infobipMessage.ReceivedAt
+
+		if text == "" {
+			continue
+		}
+
+		date := time.Now()
+		if dateString != "" {
+			date, err = time.Parse("2006-01-02T15:04:05.999999999-0700", dateString)
+			if err != nil {
+				return nil, courier.WriteError(w, r, err)
+			}
+		}
+
+		// create our URN
+		urn := urns.NewTelURNForCountry(infobipMessage.From, channel.Country())
+
+		// build our infobipMessage
+		msg := h.Backend().NewIncomingMsg(channel, urn, text).WithReceivedOn(date).WithExternalID(messageID)
+
+		// and write it
+		err = h.Backend().WriteMsg(msg)
+		if err != nil {
+			return nil, courier.WriteError(w, r, err)
+		}
+		msgs = append(msgs, msg)
+
+	}
+
+	if len(msgs) == 0 {
+		return nil, courier.WriteError(w, r, errors.New("no message found"))
```

Comment:
In what cases would this occur? Why is this an error but having no list an ignore?

Reply:
If we had a list of messages without text

Reply:
also we have a list some could be missing the text and others valid to create a message, currently we create as much msgs as we can and ignore the invalid dicts

Reply:
Ya, understood, but why is it an error here but an ignore on line 89? If we get a payload that has no messages to deal with then we should be consistent in either returning an error or just saying that's fine and we ignore it.

Reply:
Because we had the count greater than on and no text for the message like https://github.com/nyaruka/courier/pull/54/files#diff-abfdf7a6298b5a6cbf9404b33e084a46R68 so we ignored that in 99

if not message is created for the request then we return an error here

Reply:
But why? Why is that case an error (empty text) but having no messages an ignore?

Reply:
Okay I can ignore both cases then